File: problems/1001-1100/1095_find-in-mountain-array/provided/javascript/oracle.js
// Problem-provided oracle (MountainArray), JavaScript side. Evaluated
// with every submission by the judge; never editable in the editor.
// Constructed from the case state: the sequence's values (generic
// array) and the query budget. Integers may arrive as BigInt for
// exactness.
class MountainArray {
    constructor(construction, budget) {
        this.values = construction[0].map((item) => Number(item));
        this.budget = Number(budget);
    }

    get(index) {
        if (this.budget <= 0) {
            throw new Error("MountainArray query budget exhausted");
        }
        this.budget -= 1;
        if (index < 0 || index >= this.values.length) {
            throw new Error("MountainArray index out of range");
        }
        return this.values[index];
    }

    length() {
        return this.values.length;
    }
}

File: problems/1001-1100/1095_find-in-mountain-array/provided/rust/oracle.rs
// Problem-provided oracle (MountainArray), Rust side. Assembled into
// every submission's crate by the judge; never editable in the editor.
// Constructed from the case state: the sequence's values as generic
// values, then the query budget.
#[allow(dead_code)]
pub struct MountainArray {
    values: Vec<i32>,
    budget: i64,
}

impl MountainArray {
    pub fn new(construction: &[OjValue], budget: i64) -> Self {
        let items = match construction.first() {
            Some(OjValue::Array(items)) => items.clone(),
            _ => panic!("MountainArray values must be an array"),
        };
        let mut values = Vec::with_capacity(items.len());
        for item in items {
            match item {
                OjValue::Int(v) => values.push(v as i32),
                _ => panic!("MountainArray values must be integers"),
            }
        }
        MountainArray { values, budget }
    }

    pub fn get(&mut self, index: i32) -> i32 {
        if self.budget <= 0 {
            panic!("MountainArray query budget exhausted");
        }
        self.budget -= 1;
        if index < 0 || (index as usize) >= self.values.len() {
            panic!("MountainArray index out of range");
        }
        self.values[index as usize]
    }

    pub fn length(&self) -> i32 {
        self.values.len() as i32
    }
}
